lib/core/unittest/valgrind.supp

```diff
@@ -0,0 +1,965 @@
+   fun:_ZN5boost3log11v2_mt_posixlsINS1_20basic_record_ostreamIcEENSt3__112basic_stringIcNS5_11char_traitsIcEENS5_9allocatorIcEEEEEENS1_3aux38enable_record_ostream_generic_operatorIT_T0_Lb0ERSE_E4typeESG_RKSF_
+   fun:_ZN2ml5maths7CSignal3fftERNSt3__16vectorINS2_7complexIdEENS2_9allocatorIS5_EEEE
+   fun:_ZN2ml5maths7CSignal16autocorrelationsERKNSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS2_9allocatorIS8_EEEERNS3_IdNS9_IdEEEE
+   fun:_ZN2ml5maths12_GLOBAL__N_132mostSignificantPeriodicComponentENSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS2_9allocatorIS8_EEEE
+   fun:_ZN2ml5maths14testForPeriodsERKNS0_33CPeriodicityHypothesisTestsConfigEllRKNSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS4_9allocatorISA_EEEE
+   fun:_ZN2ml5maths30CTimeSeriesDecompositionDetail16CPeriodicityTest4testERKNS1_9SAddValueE
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Value8
+   fun:_ZNSt3__115basic_streambufIcNS_11char_traitsIcEEE6xsputnEPKcl
+   fun:_ZNSt3__115basic_streambufIcNS_11char_traitsIcEEE5sputnEPKcl
+   fun:_ZNSt3__116__pad_and_outputIcNS_11char_traitsIcEEEENS_19ostreambuf_iteratorIT_T0_EES6_PKS4_S8_S8_RNS_8ios_baseES4_
+   fun:_ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m
+   fun:_ZNSt3__1lsIcNS_11char_traitsIcEENS_9allocatorIcEEEERNS_13basic_ostreamIT_T0_EES9_RKNS_12basic_stringIS6_S7_T1_EE
+   fun:_ZN2ml4core17CContainerPrinter5printINSt3__111__wrap_iterIPKNS3_7complexIdEEEEEENS3_12basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEET_SG_
+   fun:_ZN2ml4core17CContainerPrinter5printINSt3__16vectorINS3_7complexIdEENS3_9allocatorIS6_EEEEEENS3_12basic_stringIcNS3_11char_traitsIcEENS7_IcEEEERKT_
+   fun:_ZN2ml5maths7CSignal3fftERNSt3__16vectorINS2_7complexIdEENS2_9allocatorIS5_EEEE
+   fun:_ZN2ml5maths7CSignal16autocorrelationsERKNSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS2_9allocatorIS8_EEEERNS3_IdNS9_IdEEEE
+   fun:_ZN2ml5maths12_GLOBAL__N_132mostSignificantPeriodicComponentENSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS2_9allocatorIS8_EEEE
+   fun:_ZN2ml5maths14testForPeriodsERKNS0_33CPeriodicityHypothesisTestsConfigEllRKNSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS4_9allocatorISA_EEEE
+   fun:_ZN2ml5maths30CTimeSeriesDecompositionDetail16CPeriodicityTest4testERKNS1_9SAddValueE
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Cond
+   fun:_ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE11__recommendEm
+   fun:_ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE6__initIPcEENS_9enable_ifIXsr21__is_forward_iteratorIT_EE5valueEvE4typeES9_S9_
+   fun:_ZNSt3__112basi
```

Comment:
Didn't we manage to avoid this problem in #852?

Reply:
nope, #852 was targeted specific Valgrind errors thrown up by `CSignal`. I suppressed this error because it appears to be out of our control. I'll review that decision.

Reply:
The unmangled function name in this one is `std::__1::complex<double>& std::__1::complex<double>::operator*=<double>(std::__1::complex<double> const&` (from `llvm-cxxfilt`).  That is different to all the ones related to `ostream`s, and it was `operator*=` for `complex` that was showing up in #845.

Reply:
Ah, ok. Thanks!